src/gsy_e/models/market/future.py

```diff
@@ -54,9 +74,46 @@ def __init__(self, bc: Optional[NonBlockchainInterface] = None,
                          readonly=readonly, grid_fee_type=grid_fee_type,
                          grid_fees=grid_fees, name=name, in_sim_duration=True)
 
-        self.slot_bid_mapping: Dict[DateTime, List[Bid]] = {}
-        self.slot_offer_mapping: Dict[DateTime, List[Offer]] = {}
-        self.slot_trade_mapping: Dict[DateTime, List[Trade]] = {}
+        self._offers = FutureOrders()
+        self._bids = FutureOrders()
```

Comment:
Can you leave a small clarification comment here, that the super().__init__() should always precede these, in order to not mistakenly move them before the constructor in the future? If the constructor was moved under these, they would be replaced by normal Python dicts (since they would be defined by the base class constructor)

Reply:
@spyrostz actually these two lines can be completely removed, because this class defines the property setter and once we try to assign the value in the parent class `self.offers = {}` it will resort to the setter which in its turn will transform the dict object into a FutureOrders object, so we're safe
I left these two here in the constructor in order to make it easier to understand the changes and to make sure that these private members are defined in the __init__, but if you believe I should remove/add explicit comment, let me know

Reply:
The problem is though that other market types can operate directly on `_bids` and `_offers` members (IIRC bid() or offer() methods do this `self._bids[bid.id] = bid`). Therefore even if the `bids` / `offers` properties are overridden, they would serve the original `_bids` and `_offers` in this case, which would be normal dicts. Please correct me if I have not seen something that I should have here.

Reply:
No other market type have the `_bids` nor the `_offers` members, AFAIS these members are indeed defined in strategies and market agents, so they should not interfere (I suppose)

Reply:
You are right, of course! I obviously confused it with the strategy members, and indeed since the members are `bids` and `offers` they can work even without it. Thanks for bearing with me !

Reply:
Thanks a lot! I will be moving this PR forward then